code/modules/projectiles/pins.dm

```diff
@@ -0,0 +1,201 @@
+/*
+Alright boys, Firing pins. hopefully with minimal shitcode.
+"pin_auth(mob/living/user)" is the check to see if it fires, put the snowflake code here. return one to fire, zero to flop. ezpz
+
+Firing pins as a rule can't be removed without replacing them, blame a really shitty mechanism for it by NT or something idk, this is to stop people from just taking pins from like a capgun or something.
+*/
+
+/obj/item/device/firing_pin
+	name = "electronic firing pin"
+	desc = "A small authentication device, to be inserted into a firearm receiver to allow operation. NT safety regulations require all new designs to incorporate one."
+	icon = 'icons/obj/firingpins.dmi'
+	icon_state = "firing_pin"
+	item_state = "pen"
+	origin_tech = list(TECH_MATERIAL = 2, TECH_COMBAT = 2)
+	flags = CONDUCT
+	w_class = 1
+	attack_verb = list("poked")
+	var/emagged = FALSE
+	var/fail_message = "<span class='warning'>INVALID USER.</span>"
+	var/selfdestruct = 0 // Explode when user check is failed.
+	var/force_replace = 0 // Can forcefully replace other pins.
+	var/pin_removeable = 0 // Can be replaced by any pin.
+	var/obj/item/weapon/gun/gun
+
+
+
+/obj/item/device/firing_pin/New(newloc)
+	..()
+	if(istype(newloc, /obj/item/weapon/gun))
+		gun = newloc
+
+/obj/item/device/firing_pin/afterattack(atom/target, mob/user, proximity_flag)
+	if(proximity_flag)
+		if(istype(target, /obj/item/weapon/gun))
+			var/obj/item/weapon/gun/G = target
+			if(G.pin && (force_replace || G.pin.pin_removeable))
+				G.pin.forceMove(get_turf(G))
+				G.pin.gun_remove(user)
+				to_chat(user, "<span class ='notice'>You remove [G]'s old pin.</span>")
+
+			if(!G.pin)
+				gun_insert(user, G)
+				to_chat(user, "<span class ='notice'>You insert [src] into [G].</span>")
+			else
+				to_chat(user, "<span class ='notice'>This firearm already has a firing pin installed.</span>")
+
+/obj/item/device/firing_pin/emag_act(mob/user)
+	if(!emagged)
+		emagged = TRUE
+		to_chat(user, "<span class='notice'>You override the authentication mechanism.</span>")
+
+/obj/item/device/firing_pin/proc/gun_insert(mob/living/user, obj/item/weapon/gun/G)
+	gun = G
+	user.drop_from_inventory(src)
+	forceMove(gun)
+	gun.pin = src
+	return
+
+/obj/item/device/firing_pin/proc/gun_remove(mob/living/user)
+	gun.pin = null
```

Comment:
Where does the old pin go?

Reply:
It used to stay in the gun's contents. Now it gets qdel'd